agent/agents/supervisor/supervisor.go

```diff
@@ -186,6 +186,11 @@ func (s *Supervisor) AgentLogByID(id string) ([]string, uint) {
 	return nil, 0
 }
 
+// ClearChanges returns channel with Agent's state changes.
+func (s *Supervisor) ClearChanges() {
+	s.changes = make(chan *agentpb.StateChangedRequest, 100)
```

Comment:
it should be locked to be safe from simultaneous access

Reply:
The read below in the `Changes` func should also have a lock.

Reply:
Done

Reply:
don't forget to close the old channel if it's being used by someone.

Reply:
the channel is used here https://github.com/percona/pmm/pull/1536/files#diff-e84ef06ecfdbc9bd2dfd4ca3aad8a21039e5c5c1766386735710c724ece67a05R446 as well